pkg/trace/api/api.go

```diff
@@ -210,9 +212,19 @@ func (r *HTTPReceiver) replyTraces(v Version, w http.ResponseWriter) {
 
 // handleTraces knows how to handle a bunch of traces
 func (r *HTTPReceiver) handleTraces(v Version, w http.ResponseWriter, req *http.Request) {
+	if atomic.LoadInt64(&r.refuse) == 1 {
+		// using too much memory
+		io.Copy(ioutil.Discard, req.Body)
+		w.WriteHeader(http.StatusNotAcceptable)
+		io.WriteString(w, "request rejected; trace-agent is past memory threshold (apm_config.max_memory)")
+		metrics.Gauge("datadog.trace_agent.receiver.refused", 1, []string{"reason:mem"}, 1)
```

Comment:
I think it might make more sense to make this a count? That way we can know exactly how many requests were refused and we can conceptually turn it into a gauge by just checking for which periods the count was > 0.

Reply:
You're right. It was meant to be a count 😆 This is a mistake.